### ISSUE TITLE
affine_transform[b, W, x] VS b + W* x

### ISSUE THREAD

**Issue by jcyk** [clab/dynet#400]
The docs say affine_transform(xs) Return an expression equal to: xs[0] + xs[1]*xs[2] + xs[3]*xs[4] + ...
But it's not absolutely right. The dims of two outputs are not equal.
For example, the following code
```
import dynet as dy
m = dy.Model()
_W = m.add_parameters((4,3))
_b = m.add_parameters(4)
x = dy.inputVector([1,2,3])
W = dy.parameter(_W)
b = dy.parameter(_b)
y = b + W * x
z = dy.affine_transform([b, W , x])
print y.dim(), z.dim()
```
will print  `((4,), 1) ((4, 1), 1)`

**Comment by neubig** [comment]
This is a good point. A 4x1 matrix is essentially a length-4 vector, and DyNet often treats them interchangeably, but I agree that the behavior should be consistent.

**Comment by jcyk** [comment]
It is kind of problematic if we use the `npvalue` to access the values. Some numpy operations are not interchangeable for these two, e.g. `argsort()`.

**Comment by pmichel31415** [comment]
I'm willing to fix this but what should the behaviour be then?

**Comment by neubig** [comment]
@pmichel31415 Let's say the dimensions of the output of an affine transform should always be the same as the bias (in this case `b`). I think this will fix the problem.

**Comment by neubig** [comment]
Fixed by #430
